File: samples/server/petstore/springboot-paged-model/src/main/java/org/openapitools/model/PetSort.java
package org.openapitools.model;

import java.net.URI;
import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonValue;
import org.openapitools.jackson.nullable.JsonNullable;
import java.io.Serializable;
import java.time.OffsetDateTime;
import jakarta.validation.Valid;
import jakarta.validation.constraints.*;


import java.util.*;
import jakarta.annotation.Generated;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Gets or Sets PetSort
 */

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen", comments = "Generator version: 7.22.0-SNAPSHOT")
public enum PetSort implements Serializable {
  
  ID_ASC("id,asc"),
  
  ID_DESC("id,desc"),
  
  NAME_ASC("name,asc"),
  
  NAME_DESC("name,desc");

  private final String value;

  PetSort(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static PetSort fromValue(String value) {
    for (PetSort b : PetSort.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}

File: samples/server/petstore/kotlin-springboot-sort-validation/src/main/kotlin/org/openapitools/model/PetSort.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonValue
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.DecimalMax
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import jakarta.validation.Valid

/**
* 
* Values: idCommaAsc,idCommaDesc,createdAtCommaAsc,createdAtCommaDesc
*/
enum class PetSort(@get:JsonValue val value: kotlin.String) : java.io.Serializable {

    idCommaAsc("id,asc"),
    idCommaDesc("id,desc"),
    createdAtCommaAsc("createdAt,asc"),
    createdAtCommaDesc("createdAt,desc");

    companion object {
        @JvmStatic
        @JsonCreator
        fun forValue(value: kotlin.String): PetSort {
                return values().firstOrNull{it -> it.value == value}
                    ?: throw IllegalArgumentException("Unexpected value '$value' for enum 'PetSort'")
        }
    }
}

